### DISCUSSION TITLE
Code Doesn't Work on C++

### DISCUSSION THREAD

**Discussion by chokostica** [community/community#66364] [Programming Help] [answered]
### Select Topic Area

Question

### Body

[csma-cd-protocol-main.zip](https://github.com/community/community/files/12550672/csma-cd-protocol-main.zip)

I cloned the attached codes into Visual Studio C++. I realized there is no .sln file and hence I cannot run the code. Please help me with steps how I can execute the codes in the attached zip on Visual studio C++.

**Comment by aregtech** [comment]
It would be better if you place the link of the cloned repo.

In case if you want to compile with Visual Studio, but don't have solution file _.sln_, you can still compile codes using CMake build tool.
As an example to compile C++ project in MS Visual Studio using CMake, you can read [this](https://github.com/aregtech/areg-sdk#build-with-visual-studio-code) small generic description. In the similar way you can compile in Visual Studio Code.

**Reply by chokostica** [reply]
See below link of the cloned repo

https://github.com/Bilal-Mhmd/csma-cd-protocol

**Reply by aregtech** [reply]
Ah, OK. Here by default you cannot compile neither with MS Visual Studio, nor with CMake. You need manually create solution and project files for MSVC or manually create _CMakeLists.txt_.

Since it is a console application, I would suggest to create empty console project in MSVS, copy the sources to the project folder and and include them in the project.

**Reply by chokostica** [reply]
Thanks @aregtech for your response.

However I do not know how to do the suggestions here:
**_Since it is a console application, I would suggest to create empty console project in MSVS, copy the sources to the project folder and and include them in the project._**

If its possible to do a video guide? Will be glad. Thanks!

**Reply by aregtech** [reply, marked as answer]
Since the task is indeed very simple, i made changes for you [in this forked repo](https://github.com/aregtech/csma-cd-protocol). 🙂
You can clone, open [**csma-cd-protocol.sln**](https://github.com/aregtech/csma-cd-protocol/blob/main/csma-cd-protocol.sln) in your MS Visual Studio, compile and run. It gives a warning about wrong conversion. I would recommend to resolve it and make sure that all variables are initialized before they are used.

**Reply by chokostica** [reply]
Wow!
Many thanks @aregtech. This works for me, wonderfully well!!!

But then again, i was expecting the code to run based on the algorithm on ReadMe file. But no, it just gives a static output, which I feel does not align. Don't kill me, I worry you too much. I just want to get it right. It's an academic research...

**Reply by aregtech** [reply]
Well, this is another problem, not related with MS Visual Studio / compilation 🙂
If you say that the code does not work as it is described in the Readme, then there is a bug.
Now, since you can compile with MS Visual Studio, by pressing `F5` you can start debug the application, find and fix the bug.
